File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/admin/user/UserManagementUiState.kt
package com.hhp227.concafe.presentation.main.admin.user

import com.hhp227.concafe.domain.model.AdminUserFilter
import com.hhp227.concafe.domain.model.User

data class UserManagementUiState(
    val selectedFilter: AdminUserFilter = AdminUserFilter.CAFE_OWNER,
    val users: List<User> = emptyList(),
    val nextCursor: String? = null,
    val canLoadMore: Boolean = false,
    val isLoading: Boolean = false,
    val isLoadingMore: Boolean = false,
    val infoMessage: String? = null
) {
    val filterChips: List<UserManagementFilterChip>
        get() = AdminUserFilter.entries.map { filter ->
            UserManagementFilterChip(
                filter = filter,
                label = filter.label,
                isSelected = selectedFilter == filter
            )
        }
}

data class UserManagementFilterChip(
    val filter: AdminUserFilter,
    val label: String,
    val isSelected: Boolean
)

val AdminUserFilter.label: String
    get() = when (this) {
        AdminUserFilter.CAFE_OWNER -> "카페 운영자"
        AdminUserFilter.BANNED -> "차단 사용자"
    }

File: iosApp/iosApp/Presentation/Main/Admin/User/UserManagementUiState.swift
import Foundation
import Shared

struct UserManagementUiState {
    var selectedFilter: AdminUserFilter = .cafeOwner
    var users: [User] = []
    var nextCursor: String? = nil
    var canLoadMore: Bool = false
    var isLoading: Bool = false
    var isLoadingMore: Bool = false
    var infoMessage: String? = nil

    var filterChips: [UserManagementFilterChip] {
        [AdminUserFilter.cafeOwner, AdminUserFilter.banned].map { filter in
            UserManagementFilterChip(
                filter: filter,
                label: filter.label,
                isSelected: selectedFilter == filter
            )
        }
    }
}

struct UserManagementFilterChip: Identifiable {
    var id: AdminUserFilter { filter }
    let filter: AdminUserFilter
    let label: String
    let isSelected: Bool
}

extension AdminUserFilter {
    var label: String {
        switch self {
        case .cafeOwner: return "카페 운영자"
        case .banned: return "차단 사용자"
        default: return "사용자"
        }
    }
}
